### PR TITLE
All comments load balancing

### PR DESCRIPTION
**Pull request by rich-nguyen**
Fixes bugs in the all discussion feature. The all comments should load whenever truncation is removed, and the load needs to be throttled to reduce cpu and i/o load.

### CODE REVIEW

File: static/src/javascripts/projects/common/modules/discussion/whole-discussion.js [hunk 1]
```diff
@@ -17,7 +17,36 @@ define([
...
+    function runConcurrently (workFunction, items) {
+
+        return new Promise(function (resolve) {
+            var queue = [];
+
+            function onComplete () {
+                if (queue.length) {
+                    start(queue.shift());
+                } else {
+                    resolve();
```
Comment (mattosborn) [inline static/src/javascripts/projects/common/modules/discussion/whole-discussion.js]: will resolve before all work is complete when `concurrentLimit > 1`

File: static/src/javascripts/projects/common/modules/discussion/whole-discussion.js [hunk 2]
```diff
@@ -17,7 +17,31 @@ define([
...
+            }
+
+            function start(item) {
+                workFunction.call(null, item).then(onComplete, onComplete);
+            }
+
+            var initialItems = items.splice(0, concurrentLimit),
+                queue = items;
+
+            _.forEach(initialItems, function (item) { start(item); });
```
Comment (mattosborn) [inline static/src/javascripts/projects/common/modules/discussion/whole-discussion.js]: `_.forEach(initialItems, start);`

File: static/src/javascripts/projects/common/modules/discussion/whole-discussion.js (hunk 1 above)
Comment (rich-nguyen) [inline static/src/javascripts/projects/common/modules/discussion/whole-discussion.js]: got it. ive put back in the workers count

**Comment by rich-nguyen** [comment]
Top feedback!

**Comment by rich-nguyen** [comment]
:diamonds:

**Pull request merged**